### PR TITLE
Configuration object

### PR DESCRIPTION
**Pull request by programminghoch10**
The client side configuration object, as defined in the backend in Gamify-IT/regexgame-backend#6
and support for specifiying an configuration id, which will be fetched from the backend when the game is started.

Depends on Gamify-IT/regexgame-backend#6
Closes Gamify-IT/issues#468

### CODE REVIEW

File: src/model/RegexStructure.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  CHARACTER_CLASS,
+  CHARACTER_CLASS_RANGE,
+  NEGATED_CHARACTER_CLASS,
+  SINGLE_CHARACTER_MATCH,
+
+}
+
+function getRegexStructureFromString(string: string): RegexStructure {
+  if (!Object.values(RegexStructure).includes(string)) throw "this regex structure doesnt exist"
+  return Object.entries(RegexStructure).find(entry => entry[0] === string)![1] as RegexStructure
```
Comment (delvh) [inline src/model/RegexStructure.ts:15]: … I have absolutely no idea what exactly you are doing here.

File: src/model/GameConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  constructor(id: string, allowedRegexStructures: Set<RegexStructure>) {
+    this.id = id
+    this.allowedRegexStructures = allowedRegexStructures
+  }
+
+  static fromJSON(json: string): GameConfiguration {
+    const parsedJSON = JSON.parse(json)
+    if (!parsedJSON.id) throw "missing id"
+    if (!parsedJSON.allowedRegexStructures) throw "missing allowed regex structures"
+    if (!(parsedJSON.allowedRegexStructures instanceof Array)) throw "regex structures invali"
```
Comment (delvh) [inline src/model/GameConfiguration.ts:21]:
```suggestion
    if (!(parsedJSON.allowedRegexStructures instanceof Array)) throw "regex structures invalid. Expected array but got " + parsedJSON.allowedRegexStructures
```

File: src/startpage.css [hunk 3]
```diff
@@ -19,28 +19,89 @@
...
 
-.button.green:hover {
-  color: #0f0;
+#startpage .button:hover {
   scale: 1.1;
+  filter: grayscale(0);
+}
+
+.button.green {
+  color: #0f0;
```
Comment (delvh) [inline src/startpage.css:34]:
```suggestion
  color: green;
```

File: src/model/RegexStructure.ts (hunk 1 above)
Comment (programminghoch10) [reply src/model/RegexStructure.ts:15]: Exactly what the method name says: Converting a string to the regex structure equivalent.
So "`CHARACTER_CLASS`" will map to `0`.

File: src/startpage.css (hunk 3 above)
Comment (programminghoch10) [reply src/startpage.css:34]: green is actually `#080` and `#0f0` would be lime.
i think for readability better leave it as is, as we do it in the same format on red too.

**Pull request merged**